common/views/components/OptIn/OptIn.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import { useState, useEffect, useContext } from 'react';
+import styled from 'styled-components';
+import cookie from 'cookie-cutter';
+import TogglesContext from '@weco/common/views/components/TogglesContext/TogglesContext';
+import { london } from '@weco/common/utils/format-date';
+import { font, classNames } from '@weco/common/utils/classnames';
+import Space from '@weco/common/views/components/styled/Space';
+import Icon from '../Icon/Icon';
+
+const OptInContainer = styled(Space).attrs(props => ({
+  as: 'div',
+  h: { size: 'l', properties: ['margin-bottom'] },
+  className: font('hnl', 5),
+}))`
+  background: ${props => props.theme.colors.purple};
+  color: ${props => props.theme.colors.white};
+  position: relative;
+  &:before {
+    content: '';
+    position: absolute;
+    left: 0;
+    top: 100%;
+    border-bottom: 18px solid transparent;
+    border-left: 16px solid ${props => props.theme.colors.purple};
+  }
+`;
+
+const OptInControl = styled.button.attrs({
+  className: classNames({
+    'plain-button no-margin no-padding flex-inline flex--v-center ': true,
+  }),
+})`
+  text-decoration: underline;
+  color: ${props => props.theme.colors.white};
+  cursor: pointer;
+  position: ${props => (props.optOut ? 'absolute' : 'static')};
+  top: ${props => `${props.theme.spacingUnit}px`};
+  right: ${props => `${props.theme.spacingUnit}px`};
+  &:focus,
+  &:hover {
+    text-decoration: none;
+  }
+`;
+
+const HiddenText = styled.span.attrs({
+  className: 'visually-hidden',
+})``;
+
+type Props = {|
+  text: {|
+    defaultMessage: string[],
+    optedInMessage: string[],
+    optInCTA: string,
+    optOutCTA: string,
+  |},
+  cookieName: string,
+|};
+
+const OptIn = ({ text, cookieName }: Props) => {
+  const [shouldRender, setShouldRender] = useState(false);
+  const [cookieValue, setCookieValue] = useState(false);
```

Comment:
Is this more about the active state rather than cookie? The cookie is just a means of setting / getting that, and it's not really the cookie value (one being Boolean the other string).